src/api/objects/ObjectAPI.js

```diff
@@ -528,7 +527,9 @@ export default class ObjectAPI {
                 && provider.observe !== undefined
                 && this.SYNCHRONIZED_OBJECT_TYPES.includes(object.type)) {
                 let unobserve = provider.observe(identifier, (updatedModel) => {
-                    if (updatedModel.persisted > mutableObject.modified) {
+                    // modified can sometimes be undefined, so make it 0 in this case
+                    const mutableObjectModification = mutableObject.modified ?? 0;
```

Comment:
I can't think of any reasonable way that this could cause a bug in this case, but when I'm dealing with time I tend to use `Number.NEGATIVE_INFINITY` or `Number.MIN_SAFE_INTEGER` instead of `0` for this purpose as any date (even those before 1970) will evaluate to true in this case.

Reply:
I could see non-wall clock time being weird and having strange numbers! Good catch